src/main/java/com/sap/oss/phosphor/fosstars/data/github/SecuregoDataProvider.java

```diff
@@ -0,0 +1,263 @@
+   */
+  private static boolean securegoConfigExists(Object step) {
+    return Optional.ofNullable(step)
+        .filter(String.class::isInstance)
+        .map(String.class::cast)
+        .filter(run -> RUN_STEP_SECUREGO_REGEX_PATTERN.matcher(run).matches())
+        .isPresent();
+  }
+
+  /**
+   * Checks if a step contains config to run Securego scans with rules.
+   *
+   * @param step The step to be checked.
+   * @return True if the step contains config to run Securego scan with rules, false otherwise.
+   */
+  private static boolean runsSecuregoWithRules(Object step) {
+    return Optional.ofNullable(step)
+        .filter(String.class::isInstance)
+        .map(String.class::cast)
+        .filter(run -> containsAnyRulesPattern(run))
+        .isPresent();
+  }
+
+  /**
+   * Checks if a step contains argument config to run Securego scans with rules.
+   *
+   * @param step The step to be checked.
+   * @return True if the step contains arguments to run Securego scan with rules, false otherwise.
+   */
+  private static boolean hasArgsWithRules(Object step) {
+    return Optional.ofNullable(step)
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .map(with -> with.get("args"))
+        .filter(String.class::isInstance)
+        .map(String.class::cast)
+        .filter(args -> containsAnyRulesPattern(args))
+        .isPresent();
+  }
+
+  /**
+   * Checks if a step contains any pattern to run Securego with rules.
+   *
+   * @param stepInfo The string info of a step
+   * @return True if the step matches any pattern to run Securego scan with rules, false otherwise.
+   */
+  private static boolean containsAnyRulesPattern(String stepInfo) {
+    return RUN_STEP_SECUREGO_WITH_INCLUDE_REGEX_PATTERN.matcher(stepInfo).matches()
+        || RUN_STEP_SECUREGO_WITH_EXCLUDE_REGEX_PATTERN.matcher(stepInfo).matches();
+  }
+
+  @Override
+  public boolean triggersScan(Map<?, ?> githubAction) {
+    return Optional.ofNullable(githubAction.get("jobs"))
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .map(jobs -> jobs.values())
+        .filter(Iterable.class::isInstance)
+        .map(Iterable.class::cast)
+        .map(SecuregoDataProvider::scanJobs)
+        .orElse(false);
+  }
+
+  /**
+   * Checks if any step in a collection of jobs triggers a Securego scan.
+   *
+   * @param jobs The collection of jobs from GitHub action.
+   * @return True if a step triggers a Securego scan, false otherwise.
+   */
+  private static boolean scanJobs(Iterable<?> jobs) {
+    return IteratorUtils.toList(jobs.iterator()).stream()
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .map(job -> job.get("steps"))
+        .filter(Iterable.class::isInstance)
+        .map(Iterable.class::cast)
+        .filter(SecuregoDataProvider::hasSecuregoStep)
+        .findAny()
+        .isPresent();
```

Comment:
there is the option of using `matchAny()` which returns true or false on any single match.